shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -109,7 +143,8 @@ - (FlutterViewController*)rootFlutterViewController {
   return nil;
 }
 
-- (void)applicationDidEnterBackground:(UIApplication*)application {
+- (void)handleDidEnterBackground:(NSNotification*)notification {
+  UIApplication* application = [UIApplication sharedApplication];
 #if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
```

Comment:
Sorry this is unrelated to your PR but is affected by it. 

I added this thing in as a hack way back in the days but didn't realize this moved into this file. It shouldn't be here (i.e. if I add to app, my app shouldn't magically start spawning background tasks on going to background because Flutter engine made a delegate which adds observers, even in debug).

Let's move this to to FlutterAppDelegate to reduce the scope a bit.

Reply:
Feel free to do this in a PR before or after